### ISSUE TITLE
when GPS is selected device only, it takes 100 seconds to get the location

### ISSUE THREAD

**Issue by shreyanshjain1991** [david-serrano/locationprovider#3]

**Comment by shreyanshjain1991** [comment]
but if we switch off the app and turn on again, it fetches the location immediately, the problem is when we return from the gps settings page after making the request, it takes almost 100 seconds to get the location

**Comment by david-serrano** [comment]
Hi, thanks for submitting this! Could you give me some more details about your implementation? For example, do you restart the provider when you're coming back into the app? something like this: 
`override fun onWindowFocus(...) {
 lProvider.startUpdates()
}`

**Comment by shreyanshjain1991** [comment]
not like this, but in onResume of the activity, I make request to location again.

**Comment by shreyanshjain1991** [comment]
I call this method : locationProvider.requestLocation(); in onResume()

**Comment by david-serrano** [comment]
yes sorry, that's the method i meant, i will have a look into this as soon as i can. Has debugging it give you any visibility as to what part of the flow hangs for 100 seconds?

**Comment by shreyanshjain1991** [comment]
only i can see in the logcat that, locationRequestStopped callback is being called soon after I do locationProvider.requestLocation(); in the onResume of the activity.
